google/cloud/grpc_options.cc

```diff
@@ -31,6 +32,12 @@ void ConfigureContext(grpc::ClientContext& context, Options const& opts) {
     context.set_compression_algorithm(
         opts.get<GrpcCompressionAlgorithmOption>());
   }
+  if (opts.has<UserIpOption>() && !opts.has<QuotaUserOption>()) {
```

Comment:
Putting this in the metadata decorator would be more consistent: https://github.com/googleapis/google-cloud-cpp/blob/4db24348ef6e3cbc5f801e5c1bee52957f42f18c/google/cloud/storage/internal/storage_metadata_decorator.cc#L879-L887

Reply:
Hmm... if anything, the metadata decorator should not set these common options and just call a common function to do it.

Reply:
> if anything, the metadata decorator should not set these common options and just call a common function to do it.

Sure, that will save us N lines of code per service.

I sort of think that the base stub should call `ConfigureContext()`, instead of the retry loops calling it.

Part of the reason why I would steer you away from `ConfigureContext()` is that I don't trust its coverage. For example, the polling loops do not use it:

https://github.com/googleapis/google-cloud-cpp/blob/4db24348ef6e3cbc5f801e5c1bee52957f42f18c/google/cloud/internal/async_polling_loop.cc#L81

Reply:
Let's merge #13935 and then I can make the changes to that function, would that work?

Reply:
SGTM

Reply:
PTAL.